src/flag_gems/ops/vector_norm.py

```diff
@@ -136,28 +263,53 @@ def vector_norm(x, ord=2, dim=None, keepdim=False, dtype=None):
         dtype = x.dtype
     if dtype not in [torch.float16, torch.float32, torch.bfloat16]:
         raise NotImplementedError(f"vector_norm not implemented for {dtype}")
-    if dim is None:
+
+    if dim is None or len(dim) == x.ndim:
```

Comment:
In this case(`full reduce`), it is not enasured that the intput is contiguous. so maybe the stage-1 kernel should consider the stride of the input.

Reply:
solved with dim_compress